src/SWP/Bundle/WebRendererBundle/Enhancer/RouteEnhancer.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Web Renderer Bundle.
+ *
+ * Copyright 2016 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2016 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\WebRendererBundle\Enhancer;
+
+use Symfony\Cmf\Component\Routing\Enhancer\RouteEnhancerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use SWP\TemplatesSystem\Gimme\Loader\LoaderInterface;
+use SWP\Bundle\WebRendererBundle\Resolver\TemplateNameResolverInterface;
+use SWP\Bundle\ContentBundle\Model\ArticleInterface;
+use Symfony\Cmf\Component\Routing\RouteObjectInterface;
+use SWP\Bundle\ContentBundle\Model\RouteInterface;
+use SWP\Bundle\WebRendererBundle\Controller\ContentController;
+
+class RouteEnhancer implements RouteEnhancerInterface
+{
+    protected $templateNameResolver;
+    protected $metaLoader;
+
+    public function __construct(TemplateNameResolverInterface $templateNameResolver, LoaderInterface $metaLoader)
+    {
+        $this->templateNameResolver = $templateNameResolver;
+        $this->metaLoader = $metaLoader;
+    }
+
+    public function enhance(array $defaults, Request $request)
+    {
+        $content = $defaults[RouteObjectInterface::CONTENT_OBJECT];
+        $defaults['_controller'] = ContentController::class . '::renderPageAction';
+        $defaults = $this->setArticleMeta($content, $request, $defaults);
+        $defaults = $this->setTemplateName($content, $defaults);
+
+        return $defaults;
```

Comment:
Yes, but i don't want to. It's easier to read when return is separate and not function result. Especially when we modify $defaults many times here.